packages/webview_flutter/webview_flutter_android/lib/src/v4/src/android_webview_controller.dart

```diff
@@ -0,0 +1,320 @@
+  })  : androidWebStorage =
+            androidWebStorage ?? android_webview.WebStorage.instance,
+        super();
+
+  /// Creates a [AndroidWebViewControllerCreationParams] instance based on [PlatformWebViewControllerCreationParams].
+  factory AndroidWebViewControllerCreationParams.fromPlatformWebViewControllerCreationParams(
+    // Recommended placeholder to prevent being broken by platform interface.
+    // ignore: avoid_unused_constructor_parameters
+    PlatformWebViewControllerCreationParams params, {
+    @visibleForTesting
+        AndroidWebViewProxy androidWebViewProxy = const AndroidWebViewProxy(),
+    @visibleForTesting android_webview.WebStorage? androidWebStorage,
+  }) {
+    return AndroidWebViewControllerCreationParams(
+      androidWebViewProxy: androidWebViewProxy,
+      androidWebStorage:
+          androidWebStorage ?? android_webview.WebStorage.instance,
+    );
+  }
+
+  /// Handles constructing objects and calling static methods for the Android WebView
+  /// native library.
+  @visibleForTesting
+  final AndroidWebViewProxy androidWebViewProxy;
+
+  /// Manages the JavaScript storage APIs provided by the [android_webview.WebView].
+  @visibleForTesting
+  final android_webview.WebStorage androidWebStorage;
+}
+
+/// Implementation of the [PlatformWebViewController] with the Android WebView API.
+class AndroidWebViewController extends PlatformWebViewController {
+  /// Creates a new [AndroidWebViewCookieManager].
+  AndroidWebViewController(PlatformWebViewControllerCreationParams params)
+      : super.implementation(params is AndroidWebViewControllerCreationParams
+            ? params
+            : AndroidWebViewControllerCreationParams
+                .fromPlatformWebViewControllerCreationParams(params));
+
+  AndroidWebViewControllerCreationParams get _androidWebViewParams =>
+      params as AndroidWebViewControllerCreationParams;
+
+  /// The native [android_webview.WebView] being controlled.
+  late final android_webview.WebView _webView = withWeakRefenceTo(this, (
+    WeakReference<AndroidWebViewController> weakReference,
+  ) {
+    return _androidWebViewParams.androidWebViewProxy.createAndroidWebView(
+      useHybridComposition: true,
+    );
+  });
+
+  /// The native [android_webview.FlutterAssetManager] allows managing assets.
+  late final android_webview.FlutterAssetManager _flutterAssetManager =
+      withWeakRefenceTo(this, (
+    WeakReference<AndroidWebViewController> weakReference,
+  ) {
+    return _androidWebViewParams.androidWebViewProxy
+        .createFlutterAssetManager();
+  });
+
+  final Map<String, AndroidJavaScriptChannelParams> _javaScriptChannelParams =
+      <String, AndroidJavaScriptChannelParams>{};
+
+  @override
+  Future<void> loadFile(
+    String absoluteFilePath,
+  ) {
+    final String url = absoluteFilePath.startsWith('file://')
+        ? absoluteFilePath
+        : 'file://$absoluteFilePath';
```

Comment:
This is actually a pre-existing bug; it should be `Uri.file(absoluteFilePath)` instead of `file://$absoluteFilePath`, so that special characters are handled correctly.

Reply:
Fixed.